src/training/data.py

```diff
@@ -30,8 +32,106 @@
 from open_clip.tokenizer import HFTokenizer
 
 
+def get_collate_fn(dataset, is_flava=False):
+    if is_flava:
```

Comment:
Very specific to FLAVA logic. Maybe the collate function should take a custom preprocessing function that it applies and returns the output. Then, in your training script you can specific the custom collate function you want.